core/app/models/spree/order_merge.rb

```diff
@@ -0,0 +1,63 @@
+module Spree
+  class OrderMerge
+    attr_accessor :destination, :source
+
+    def initialize(destination, source, user = nil)
+      @destination = destination
+      @source = source
+      merge!(user)
+    end
+
+    def merge!(user = nil)
+      source.line_items.each do |other_order_line_item|
+        next unless other_order_line_item.currency == destination.currency
+
+        # Compare the line items of the other order with mine.
+        # Make sure you allow any extensions to chime in on whether or
+        # not the extension-specific parts of the line item match
+        current_line_item = destination.line_items.detect { |my_li|
+          my_li.variant == other_order_line_item.variant &&
+            destination.line_item_comparison_hooks.all? { |hook|
```

Comment:
Avoid using {...} for multi-line blocks.<br>Align the operands of an expression in an assignment spanning multiple lines.